instructure/canvas-ios
# Offline Course Sync Files Interactor

This PR adds file download capability to course sync. 

refs: MBL-16722
affects: Student
release note: none 
test plan: 
1. Select files in the Course Sync Selector 
2. Tap Sync & wait for download
3. Check Simulator's documents folder and search for a folder called `Offline`. Downloaded files should be there. 
4. If you re-download the same files again, you should not see a loading state because all the files are downloaded already. 

edit: 
After discussion we decided to change the download logic a bit. Offline Course Sync files are downloaded to `Documents/Offline` meanwhile regular file downloads in the `FileDetailsViewController` are downloaded to the temporary folder. This implies that we'll need to prepare the view controller to handle offline state so that it will look up files in the `Offline` folder, otherwise in the temp folder. This change will be implemented in a separate PR. 

## Checklist
